File: tools/referentials/client/services/newentity.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { NewEntity } from '../../targetPackage/model/NewEntity';

@Injectable({
  providedIn: 'root'
})
export class NewEntityService extends AppRestService<NewEntity>{

  constructor(http: HttpClient) {
    super(http, "controllerName");
  }

  getNewEntities() {
    return this.getList(new HttpParams(), "entryPointName");
  }
  
   addOrUpdateNewEntity(newEntity: NewEntity) {
    return this.addOrUpdate(new HttpParams(), "entryPointNameSingular", newEntity, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: tools/referentials/backend/service/NewEntityService.java
package com.jss.osiris.modules.osiris.targetPackage.service;

import java.util.List;

import com.jss.osiris.modules.osiris.targetPackage.model.NewEntity;

public interface NewEntityService {
    public List<NewEntity> getNewEntities();

    public NewEntity getNewEntity(Integer id);

    public NewEntity addOrUpdateNewEntity(NewEntity newEntity);
}
